### ISSUE TITLE
About the training resolution

### ISSUE THREAD

**Issue by tingxueronghua** [Luo-Yihong/TDM-R1#2]
Great work and look forward to your training code release :)

Well before your official release, I want to inquire about the training resolution. I notice that in Appendix B, it is said "All experiments are conducted at 512 resolution, with GPU hours reported in A100 equivalents.". However, the inference code example on huggingface shows that you are inferring with a resolution of 1024. Does this work as expected?

**Comment by tingxueronghua** [comment]
What's more, why the huggingface requires the user to add the lora to Z-Image-Trubo? I think your model should be added to Z-image Base model?

**Comment by tingxueronghua** [comment]
<img width="1024" height="1024" alt="Image" src="https://github.com/user-attachments/assets/98846792-e8ee-439b-921a-64e3e87dca2c" />

```python
import os
os.environ["HF_ENDPOINT"] = "https://hf-mirror.com"
import torch
from diffusers import ZImagePipeline
from peft import LoraConfig, get_peft_model
def load_ema(pipeline, lora_path, adapter_name='default'):
    """Load EMA weights into the pipeline's transformer adapter"""
    pipeline.transformer.set_adapter(adapter_name)
    trainable_params = [
        p for n, p in pipeline.transformer.named_parameters()
        if adapter_name in n and p.requires_grad
    ]
    state_dict = torch.load(lora_path, map_location=pipeline.transformer.device)
    ema_params = state_dict["ema_parameters"]
    assert len(trainable_params) == len(ema_params), \
        f"Parameter count mismatch: {len(trainable_params)} vs {len(ema_params)}"
    for param, ema_param in zip(trainable_params, ema_params):
        param.data.copy_(ema_param.to(param.device))
    print(f"Loaded EMA weights for adapter '{adapter_name}' from {lora_path}")
pipeline = ZImagePipeline.from_pretrained(
    "my z image turbo path",
    torch_dtype=torch.bfloat16,
    low_cpu_mem_usage=False,
)
transformer_lora_config = LoraConfig(
    r=32,
    lora_alpha=64,
    init_lora_weights="gaussian",
    target_modules=["to_q", "to_k", "to_v", "to_out.0", "add_k_proj", "add_v_proj"],
)
pipeline.transformer = get_peft_model(
    pipeline.transformer,
    transformer_lora_config,
    adapter_name="tdmr1",
)
load_ema(
    pipeline,
    lora_path=" my path to Luo-Yihong--TDM-R1/tdmr1_zimage_ema.ckpt",
    adapter_name="tdmr1",
)
pipeline = pipeline.to("cuda")
image = pipeline(
      prompt="A beautiful lady is walking along a street",
      height=1024,
      width=1024,
      num_inference_steps=5,  # This actually results in 4 DiT forwards
      guidance_scale=0.0,
      generator=torch.Generator("cuda").manual_seed(42),
  ).images[0]

image.save("test_tdm_r.webp", lossless=True)
```


I think the result contains too many noise artifacts, which is quite hard to align with the performance of the teaser image... Could you help me with a minimal reproduce code scripts?

**Comment by Luo-Yihong** [comment]
Thank you for your attention to our work! 

For ZImage, the training resolution is 768, and we found that it generalizes well to 1024 resolution. The generator is initialized from ZImage-Turbo to save computational cost, while the reward model and score model required for training are initialized from ZImage-Base. 

For the artifact, in our internal testing, we have also observed that "noise/water drop" artifacts sometimes randomly appear in the background. We found that ZImage-Turbo actually exhibits a similar phenomenon, but RL post-training seems to have somehow amplified this issue. We will fix this problem as soon as possible and update the public model. btw, do you have any improvement suggestions for us?

**Comment by tingxueronghua** [comment]
Thanks for sharing the information! I am still learning your algorithm design.

I have also tried TDM algorithms, and I have also found the noise/water drop artifacts (along with limb distortion). GAN is helpful for this, but TDM cannot work with GAN for too long time.

I also tried flow-group-like algorithms. It will introduce more noises but I still do not figure out how to alleviate this...

**Comment by Luo-Yihong** [comment]
Thanks for your information. That was interesting for me, because I haven't observed a similar phenomenon in TDM samples before. TDM-R1 on Z-Image also first went through TDM post-training for a few iterations, where we didn't observe more "noise/water drop" artifacts compared to Z-Image-Turbo.

**Comment by tingxueronghua** [comment]
It depends on your generator's quality. If it initializes well on the few-step image generation task, the noise/water drop will be alleviated.

**Comment by tingxueronghua** [comment]
Usually, flow-matching-based t2i models (which do not do any RL-like algorithms) initialize poorly on the few-step image generation task, when working as the generator. If you do not do any optimization, they will have lots of high-frequency noises, which harm the step-distillation algorithms. If you choose a good initialization, of course, it will not have such problems.
